tests/api/console/test_cli_console_output.py

```diff
@@ -91,18 +98,35 @@ def test_phase_is_cleaned_up_on_exception(cli_console):
         pass
 
 
-def test_phase_cannot_be_indented(cli_console):
+def test_phase_with_indent(cli_console, capsys):
+    cli_console.step("Outside of Indent")
     with cli_console.indented():
-        with pytest.raises(CliConsoleNestingProhibitedError):
-            with cli_console.phase("Enter"):
-                pass
+        with cli_console.phase("Enter"):
+            pass
 
+    expected_output = dedent(
+        f"""\
+        Outside of Indent
+          Enter
+        """
+    )
 
-def test_step_cannot_be_indented(cli_console):
+    assert_output_matches(expected_output, capsys)
+
+
+def test_step_with_indent(cli_console, capsys):
```

Comment:
do we actually need this? It's a bit surprising, that wasn't the intention behind `indented`. I would expect phases can be nested, as you've done here, but steps cannot. Indented should only be used as a leaf construct, not in between phases and steps, unless I'm missing something.

Reply:
The intent was to allow everything, or else we might end up in another situation where an unexpected flow ends up raising an error, when it probably doesn't make sense for our commands to not work if we mess up the formatting a little, similar to how we decided that our metrics shouldn't blow up our commands if something goes wrong there either